pkg/nmi/iptables/iptables.go

```diff
@@ -107,3 +107,41 @@ func flushCreateCustomChainrules(ipt *iptables.IPTables, destIP, destPort, targe
 
 	return nil
 }
+
+// DeleteCustomChain deletes the custom chain aad-metadata reference from PREROUTING
+// chain and then deletes the chain aad-metadata
+func DeleteCustomChain() error {
+	ipt, err := iptables.New()
+	if err != nil {
+		return err
+	}
+	if err := deleteCustomChainInChain(ipt, tablename, "PREROUTING"); err != nil {
+		return err
+	}
+	if err := deleteCustomChain(ipt, tablename); err != nil {
+		return err
+	}
+	return nil
+}
+
+// deleteCustomChainInChain - iptables -t "table" -D "chain" -j "customchainname"
+func deleteCustomChainInChain(ipt *iptables.IPTables, table, chain string) error {
```

Comment:
Could we please make the function name more clear - perhaps something like "removeCustomChainReference" or something that like that ?